org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/ZigBeeDeviceConfigHandler.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zsmartsystems.zigbee.ZigBeeEndpoint;
+import com.zsmartsystems.zigbee.ZigBeeNode;
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+import com.zsmartsystems.zigbee.zcl.protocol.ZclDataType;
+
+/**
+ * Class to handle ZigBee device configuration. This is a helper class used by the {@link ZigBeeThingTypeHandler} for
+ * managing device configuration.
+ *
+ * @author Chris Jackson - Initial Contribution
+ *
+ */
+public class ZigBeeDeviceConfigHandler {
+    /**
+     * Our logger
+     */
+    private final Logger logger = LoggerFactory.getLogger(ZigBeeDeviceConfigHandler.class);
+
+    private final ZigBeeNode node;
+
+    public ZigBeeDeviceConfigHandler(ZigBeeNode node) {
+        this.node = node;
+    }
+
+    /**
+     * Handles a device configuration update
+     *
+     * @param configuration map of configuration parameters
+     * @return a map of updated configuration parameters
+     */
+    public Map<String, Object> handleConfigurationUpdate(Map<String, Object> configuration) {
+        Map<String, Object> updatedConfiguration = new HashMap<>();
+
+        for (Entry<String, Object> configurationParameter : configuration.entrySet()) {
+            // Since there is no ability to parameterise configuration parameters from
+            // the device perspective, we need to embed the properties into the key
+            // eg. attribute_02_in_0406_0030_18 = endpoint_direction_cluster_attribute_datatype
+            String[] cfg = configurationParameter.getKey().split("_");
+            switch (cfg[0]) {
+                case "attribute":
```

Comment:
Yes - this code is horrible. I expect it will be removed  soon as I'm hopeful that the new [configuration system](https://github.com/eclipse/smarthome/pull/5216) will have the ability to specify this as properties.
I'll update to a constant in the meantime.